### PR TITLE
Disable some SIMD tests when running under Rosetta2

### PR DESCRIPTION
**Pull request by xclerc**
This pull request disables the SIMD tests
failing when using Rosetta2 (determined
by running the test suite on an M2 box).

Rosetta2 is macOS' emulation layer that
allows one to run `amd64` binaries on
Apple Silicon boxes.

### CODE REVIEW

File: tests/simd/test_helpers.ml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        | "1" -> On
+        | _ -> Unknown
+      with _ ->
+        Unknown
+    end
+  | _ ->
+    Unknown
+
+let run_if_not_under_rosetta2 ~f =
+  match get_rosetta2_status () with
```
Comment (mshinwell) [inline tests/simd/test_helpers.ml:26]: Maybe we should print the status each time?

**Review by mshinwell — APPROVED** [review]
One minor comment

**Comment by xclerc** [comment]
Note: the tests in `scalar_ops.ml` fail with `SIGILL`
while the ones in `ops.ml` seems to fail because of
different rounding.

File: tests/simd/test_helpers.ml (hunk 1 above)
Comment (xclerc) [reply tests/simd/test_helpers.ml:26]: That would be useful, but wouldn't that
make the test not reproducible?

**Comment by xclerc** [comment]
The test suite is failing on the CI (not finding
the new module) but is OK locally. Did the
semantics of dune's `modules` stanza change
at some point?

**Comment by xclerc** [comment]
Actually, I have the same dune version locally (3.6.1).

**Comment by xclerc** [comment]
Please ignore the dune-related comments
above; the issue was an auxiliary target
which is not built on non-macOS boxes.

File: tests/simd/test_helpers.ml (hunk 1 above)
Comment (mshinwell) [reply tests/simd/test_helpers.ml:26]: Ah, true.

**Pull request merged**